src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -1114,10 +1126,67 @@ private static int GetNamespaceId(SyntaxList<MemberDeclarationSyntax> members, N
             return -1;
         }
 
+        private static bool TryNormalizeOrReduce(
+            this NameSyntax name,
+            SemanticModel semanticModel,
+            out ExpressionSyntax replacementNode,
+            out TextSpan issueSpan,
+            OptionSet optionSet,
+            CancellationToken cancellationToken)
+        {
+            if (name.TryNormalize(semanticModel, out replacementNode, out issueSpan, optionSet))
+            {
+                return true;
+            }
+
+            return name.TryReduce(semanticModel, out replacementNode, out issueSpan, optionSet, cancellationToken);
+        }
+
+        private static bool TryNormalize(
+            this NameSyntax name,
+            SemanticModel semanticModel,
+            out ExpressionSyntax replacementNode,
+            out TextSpan issueSpan,
+            OptionSet optionSet)
+        {
+            replacementNode = null;
+            issueSpan = default(TextSpan);
+
+            // if we can't resolve the symbol then nothing can be inferred
+            var symbol = SimplificationHelpers.GetOriginalSymbolInfo(semanticModel, name);
+            if (symbol == null)
+            {
+                return false;
+            }
+
+            // if not static and not already part of a `this.` expression
+            if (!symbol.IsStatic &&
+                name is SimpleNameSyntax &&
+                (name.Parent as MemberAccessExpressionSyntax)?.Expression.Kind() != SyntaxKind.ThisExpression)
```

Comment:
What about named on the right of things like ```->``` or ```?.```  I think we have syntax helpers that can ask if this is the right side of a some sort of qualified name or not.  Can you use those instead?